pkg/security/secl/model/model.go

```diff
@@ -622,6 +622,7 @@ type SetXAttrEvent struct {
 // SyscallEvent contains common fields for all the event
 type SyscallEvent struct {
 	Retval int64 `field:"retval" msg:"retval"` // Return value of the syscall
+	Async  int64 `field:"async" msg:"async"`   // True if the syscall was asynchronous
```

Comment:
```suggestion
	Async  bool `field:"async" msg:"async"`   // True if the syscall was asynchronous
```

Reply:
I think this is very interesting, but thinking about it I have 2 fears about this:
- we add async field to nearly all event types even for those where it does not make sense (bpf for example)
- we add 64 bits (8 bytes) of data to all syscall events, I'm a bit worried about the impact in the usage of the perf event queue

Reply:
Yep, while doing it I was also wondering about the same first point, even if more and more opcode will comes to iouring, theres still some events for whose it will make no sense.. We could either:
- limit it to the syscalls having iouring opcodes (removing it from the global "syscall" struct)
- make the field omitempty, which should be present only if async is true
Thoughs?

About your second point yes, for sure it's not ideal. I've not succeed to make it works with u8 due to bpf/bcc constraints. And again, even if we could make it 1 byte size, it will still be added on events without any iouring opcodes.
Maybe the generic way, which I had think would be more elegant, was a bad idea. :)

Reply:
We decided for now to:
- make the async field optional (omitempty)
- keep the way we retrieve the info from the bpf part (as an u64 :/)